### ISSUE TITLE
Clip worked for me

### ISSUE THREAD

**Issue by th0ma5w** [Dafang-Hacks/spiflasher#1]
Perhaps a needless issue, but just wanted to note here that after many tries, a clip did wind up working for me. Perhaps there is enamel or flux or something on the pins, or they need to be bent in slightly or something, but it did indeed work for me.

**Comment by EliasKotlyar** [comment]
Hello @th0ma5w ,

Thanks for your report. Which hardware were you using? And what kind of clip? I would like to provide this information in the README

Greatings
Elias

**Comment by th0ma5w** [comment]
Not sure of the reliability of these links, or the consistency of the products, I mostly just tried to find the cheapest, although I probably could've saved money had I purchased them together.

ch341a from this listing:
https://www.ebay.com/itm/253056283240

and plugged in this clip adapter into the 25 series socket and attached the ribbon and clip:
https://www.amazon.com/gp/product/B07BRSVRXV/ref=od_aui_detailpages00?ie=UTF8&psc=1

I seriously must've tried easily 20-30 times before it suddenly just started working. I'm not sure why I persisted other than it seemed like this YouTube video demonstrating it made it look easy, but even they had to try a couple of times:
https://www.youtube.com/watch?v=2Y06x1f22B0&t=616s

**Comment by th0ma5w** [comment]
Oh sorry this was the Dafang 128mb

**Comment by columbusgeek** [comment]
I'm going to try the clip and this process, thanks for the info @th0ma5w .  

BTW, which chip needs to be flashed?  I see one on the board that has the camera and the SD slot, and I see one on the board that has the female USB port.

**Comment by columbusgeek** [comment]
@th0ma5w & @EliasKotlyar 

Sorry if I'm being a pest, I just need to know which chip to flash.  I can provide pictures and help with the how-to for this project but right now I'm at a fork in the road.

**Comment by EliasKotlyar** [comment]
can you please provide the pictures?

**Comment by columbusgeek** [comment]
Yeah, I've tried to see the text but it's black on black and even with magnification it's hard to read.  

Ok, I'll bust the real camera out and take some snaps.  I just figured by this point a few people have flashed the V2s.  Guess not.  😄

**Comment by columbusgeek** [comment]
Here are some higher res pics.  I took pictures of the backs in case that was needed in the future.  Feel free to take these and use them.  
 Even using the camera it was a challenge to make the text show up.  

I'm still struggling a bit to read them as well.  Do either of them look familiar?

https://imgur.com/a/tbIubNk

**Comment by columbusgeek** [comment]
Here is the one on the board with the camera.  I cleaned it up as best I could in PS.

<img width="1625" alt="screen shot 2018-11-02 at 10 57 37 am" src="https://user-images.githubusercontent.com/20864904/47922794-4807e780-de8e-11e8-8b35-85eafd5b1a1e.png">

**Comment by columbusgeek** [comment]
Here is the one on the board with the USB.

<img width="1625" alt="screen shot 2018-11-02 at 11 01 05 am" src="https://user-images.githubusercontent.com/20864904/47922992-b187f600-de8e-11e8-9e63-69b34790a5bf.png">

**Comment by janvotava** [comment]
Clip worked for me on Xiaofang 1S, it was just pain to make it stay there and required a lot of tries.

**Comment by hamid-elaosta** [comment]
I tried a clip on the Neos Smartcam, it does NOT work.

De-soldering worked fine, I accidently ripped off the RESET pad with the clip so soldered it to the nearby resistor with Kynar and it's fine.

**Comment by th0ma5w** [comment]
I had yet another one of these I messed up... not sure why but the flashing of the boot loader via the command line on the device just doesn't seem to work right. Anyway... couldn't get the clip again to work, but put a very small drop of hand sanitizer (no moisturizer) on a napkin and rubbed it on all the pins, and then dried it thoroughly and waited about a minute. Then it worked perfectly.

**Comment by thib3113** [comment]
Hello all,

First, thank you for the informations . I'm just here to share my experience .

I'm a newby with this, I doesn't have soldering skills (or really bad), and never flash a binary ...

So, to start, I buy things mentionned in one of the first comments : 
 - ch341a + clip, I take this one : https://www.amazon.fr/gp/product/B07SNTL5V6/ref=ppx_yo_dt_b_asin_title_o02_s00?ie=UTF8&psc=1
 - a raspberry pi (because I've one, but I think it's the same with a linux computer) . (mine is an old 2B+ )

I plug the usb in the raspberry pi, in one of the usb slots, it seems to doesn't metter.
Plug the pins from the clip adaptater, on the ch341a (the usb adapter), starting on the pins from the usb, and red line on the top left .

a picture : 
![image](https://user-images.githubusercontent.com/5489218/79146861-51a3b780-7dc3-11ea-9c2c-e0227d1b645d.png)

and I run (in sudo mode) : 
`watch flashrom -p ch341a_spi -n -V -c GD25Q128C`

and try to clip the chip (red line to the top left of the chip, top is where you can see a dot on the chip platic) .

![image](https://user-images.githubusercontent.com/5489218/79147010-8fa0db80-7dc3-11ea-9b39-1b78327f50b1.png)

(the picture is just to illustrate, in fact I dissassemble all the camera to do my test, and remount it to try if it's work ... but it seems that you can clip the chip wihtout dismount totally the camera) .

the command with watch, tell me multiple times, that it can't read the chip, so each times, I remove the clip, and try to re-clip it ... and finally (after maybe 5/6 times), flashrom tell me a different message .

So I can run : 
`sudo flashrom -p ch341a_spi -n -V -c GD25Q128C --layout rom.layout --image boot -w fullflash.bin`
Waiting some minutes (maybe 5 minutes ?), and the chip is flashed :) . 

On my case, now I need to re-mount all the camera .

else, just try to power it, and see if the led become yellow ... If yes, it seems you succeed (on my case, before the led was totally black when the camera is powered) . 

And now, it's work :D 

( always on my case, for the moment, the motors doesn't work, so maybe I doesn't mount something correctly ) . 

So, thank you @th0ma5w for the things to buy, and thank you @EliasKotlyar for the command lines (and the bin, and the pictures, and all else) .

And, even if you are a newby, you can flash your camera easily :) .
